src/components/AChat/AChat.vue

```diff
@@ -78,118 +78,101 @@ export default {
+      const clientHeight = messagesRef.value.clientHeight
 
-      // Scrolled to Bottom
       if (scrollHeight - scrollTop === clientHeight) {
-        this.$emit('scroll:bottom')
+        emit('scroll:bottom')
       } else if (scrollTop === 0) {
-        // Scrolled to Top
-        // Save current `scrollHeight` to maintain scroll
-        // position when unshift new messages
-        this.currentScrollHeight = scrollHeight
-        this.$emit('scroll:top')
+        currentScrollHeight.value = scrollHeight
+        emit('scroll:top')
       }
 
-      // Save previous values of `scrollTop` and `scrollHeight`
-      // Needed for keeping the same scroll position when prepending
-      // new messages to the chat
-      this.currentScrollTop = scrollTop
-      this.currentScrollHeight = scrollHeight
+      currentScrollTop.value = scrollTop
+      currentScrollHeight.value = scrollHeight
 
-      emitScroll.call(this)
-    },
+      emitScroll()
+    }
 
-    // Fix scroll position after unshift new messages.
-    // Called from parent component.
-    maintainScrollPosition() {
-      this.$refs.messages.scrollTop =
-        this.$refs.messages.scrollHeight - this.currentScrollHeight + this.currentScrollTop
-    },
+    const maintainScrollPosition = () => {
+      if (!messagesRef.value) return
 
-    // Scroll to Bottom when new message.
-    // Called from parent component.
-    scrollToBottom() {
-      this.$refs.messages.scrollTop = this.$refs.messages.scrollHeight
-    },
+      if (isWelcome.value) {
+        messagesRef.value.scrollTop = 0
+        return
+      }
 
-    scrollTo(position) {
-      this.$refs.messages.scrollTop = position
-    },
+      messagesRef.value.scrollTop =
+        messagesRef.value.scrollHeight - currentScrollHeight.value + currentScrollTop.value
+    }
+
+    const scrollToBottom = () => {
+      if (messagesRef.value) {
+        messagesRef.value.scrollTop = messagesRef.value.scrollHeight
+      }
+    }
+
+    const scrollTo = (position: number) => {
+      if (messagesRef.value) {
+        messagesRef.value.scrollTop = position
+      }
+    }
+
+    const scrollToMessage = (index: number) => {
+      if (!messagesRef.value) return
 
-    /**
-     * Scroll to message by index, starting with the last.
-     */
-    scrollToMessage(index) {
-      const elements = this.$refs.messages.children
+      const elements = messagesRef.value.children
 
       if (!elements) return
 
-      const element = elements[elements.length - 1 - index]
+      const element = elements[elements.length - 1 - index] as HTMLElement
 
       if (element) {
-        this.$refs.messages.scrollTop = element.offsetTop - 16
+        messagesRef.value.scrollTop = element.offsetTop - 16
       } else {
-        this.scrollToBottom()
+        scrollToBottom()
       }
-    },
+    }
+
+    const scrollToMessageEasy = (index: number): Promise<boolean> => {
+      if (!messagesRef.value) return Promise.resolve(false)
```

Comment:
You can mark function as `async` and get rid of `Promise.resolve`. Check other places as well.